flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -332,7 +335,15 @@ public List<String> listTables(String databaseName)
   public CatalogTable getTable(ObjectPath tablePath)
       throws TableNotExistException, CatalogException {
     Table table = loadIcebergTable(tablePath);
-    return toCatalogTable(table);
+    Map<String, String> catalogAndTableProps = Maps.newHashMap(catalogProps);
+    catalogAndTableProps.put(FlinkCreateTableOptions.CATALOG_NAME.key(), getName());
+    catalogAndTableProps.put(
+        FlinkCreateTableOptions.CATALOG_DATABASE.key(), tablePath.getDatabaseName());
+    catalogAndTableProps.put(
+        FlinkCreateTableOptions.CATALOG_TABLE.key(), tablePath.getObjectName());
+    catalogAndTableProps.put("connector", FlinkDynamicTableFactory.FACTORY_IDENTIFIER);
+    catalogAndTableProps.putAll(table.properties());
+    return toCatalogTableWithProps(table, catalogAndTableProps);
```

Comment:
Could you help me understand why is the table properties needed to be added here?
We also send the table as a parameter. Wouldn't it be enough?

Reply:
Just thinking out loud:
- Maybe the code would be easier to read if we send only the catalogProps to the `toCatalogTableWithProps` and create a merged map when calling the Flink method
- This is somewhat suboptimal as we create an extra map

Even if we decide to follow your approach, the parameter name of the method should reflect that at the declaration of `toCatalogTableWithProps`, and maybe some comments or javadoc should be nice there for future generations 😉 

WDYT?

Reply:
Yeah merged it here , as merging later on needs an extra map.

Renamed the method to toCatalogTableWithCustomProps and modified parameter names. Hope it's more readable now.